experimental/packages/opentelemetry-instrumentation-grpc/src/grpc/index.ts

```diff
@@ -306,6 +306,11 @@ export class GrpcNativeInstrumentation extends InstrumentationBase<
             [SemanticAttributes.RPC_METHOD]: method,
             [SemanticAttributes.RPC_SERVICE]: service,
           });
+        // set peer.service from target (e.g., "dns:otel-productcatalogservice:8080") as a hint to APMs
+        const parsedUri = URI_REGEX.exec(this.getChannel().getTarget());
```

Comment:
The specification defines that [`peer.service`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-remote-service-attributes):

> SHOULD be equal to the actual `service.name` resource attribute of the remote service if any.

[`Channel#getTarget`](https://grpc.github.io/grpc/node/grpc.Channel.html#getTarget__anchor) returns the address that this client is connected to, rather than the desired `service.name`.

Reply:
haha - I agree! I was looking at values in flight from some service which happen to set it to host:port. I'll adjust. Thank you for the reference.

Reply:
Lemme see how other auto-instrumentation frameworks set this (if at all). It's a little tricky, because unless you have some out of band means for the app to specify the intended service.name (which partially defeats the auto-instrumentation model), I'm not sure how you'd know it from the sender side. In k8s environments, probably host(this.getChannel().getTarget()) is the service.name, but not guaranteed by any means. I'll dig a little.

Reply:
@legendecas - ok, I've change this to set net.peer.name and net.peer.port per [OTEL RPC conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#common-attributes). 

Per [this](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#netpeername), we always set net.peer.name regardless of whether is a DNS name or IP: "Sometimes host name is only available to instrumentation as a string which may contain DNS name or IP address. net.peer.name SHOULD be set to the available known hostname (e.g., "127.0.0.1" if connecting to an URL https://127.0.0.1.com/foo)."